plugins/amoeba/platforms/common/src/kernels/hippoNonbonded.cc

```diff
@@ -19,11 +19,21 @@ typedef struct {
 #endif
 
 #ifdef ENABLE_SHUFFLE
+#if defined(USE_HIP)
+
+#define real_shfl SHFL
+
+#else
+
 //support for 64 bit shuffles
 static __inline__ __device__ float real_shfl(float var, int srcLane) {
     return SHFL(var, srcLane);
 }
 
+static __inline__ __device__ int real_shfl(int var, int srcLane) {
+    return SHFL(var, srcLane);
+}
```

Comment:
I don't understand this function.  `real_shfl()` operates on `real`.  If we want to shuffle an `int`, that's what `SHFL()` already does.

Reply:
Good point. 
Then, I guess, `real_shfl` should be renamed/removed everywhere.
In this file you already have its overload for `int` (I don't think it's used though).
I just made it consistent with nonbonded.cu. where real_shfl is used with, for example, with `int shflatomType` and `mm_long shflobc4_bornForce`.